scripts/zones/Hazhalm_Testing_Grounds/npcs/Armoury_Crate.lua

```diff
@@ -2,23 +2,60 @@
 -- Area: Hazhalm Testing Grounds
 -- NPC: Armoury Crate
 -----------------------------------
----@type TMobEntity
+local ID = zones[xi.zone.HAZHALM_TESTING_GROUNDS]
+-----------------------------------
+---@type TNpcEntity
 local entity = {}
 
-entity.onMobInitialize = function(mob)
-    mob:setMobMod(xi.mobMod.NO_REST, 1)
-    mob:setMobMod(xi.mobMod.NO_MOVE, 1)
-    mob:setMobMod(xi.mobMod.NO_DESPAWN, 1)
-    mob:setStatus(xi.status.NORMAL)
-end
+entity.onTrigger = function(player, npc)
+    if npc:getLocalVar('[ein]tempCrate') == 0 then
+        return
+    end
 
-entity.onMobSpawn = function(mob)
-end
+    local function showTempItems(p)
+        p:startEvent(5,
+            npc:getLocalVar('[ein]tempItem1'),
+            npc:getLocalVar('[ein]tempItem2'),
+            npc:getLocalVar('[ein]tempItem3'),
+            npc:getLocalVar('[ein]tempItem4'),
+            npc:getLocalVar('[ein]tempItem5'),
+            npc:getLocalVar('[ein]tempItem6'),
+            xi.item.DUSTY_POTION, -- Retail always sends this item in the last 2 slots
+            xi.item.DUSTY_POTION
+        )
+    end
+
+    if npc:getLocalVar('opened') == 0 then
+        local tempItems = xi.einherjar.getTempItems()
 
-entity.onMobDeath = function(mob, player, optParams)
+        for i = 1, 6 do
+            npc:setLocalVar('[ein]tempItem' .. i, tempItems[i])
+        end
+
+        npcUtil.openCrate(npc, function()
+            return true
+        end)
+    end
+
+    showTempItems(player)
 end
 
-entity.onMobDespawn = function(mob)
+entity.onEventFinish = function(player, csid, option, npc)
+    if csid == 5 and option >= 1 and option <= 6 then
+        local crateItem = npc:getLocalVar('[ein]tempItem' .. option)
+
+        local quantity = bit.rshift(crateItem, 16)
+        local itemId = bit.band(crateItem, 0xFFFF)
+
+        if player:hasItem(itemId, 3) then
```

Comment:
im guessing that 3 is the inventory
`if player:hasItem(itemId, xi.inventoryLocation.TEMPITEMS) then`